builtin/providers/aws/data_source_aws_caller_identity.go

```diff
@@ -0,0 +1,40 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsCallerIdentity() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsCallerIdentityRead,
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsCallerIdentityRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient)
+
+	log.Printf("[DEBUG] Reading Caller Identity.")
+	d.SetId(time.Now().UTC().String())
+
+	if client.accountid != "" {
```

Comment:
This looks inverted... don't we want to produce this error if the account id *is* empty?